File: mifosng-android/src/main/java/com/mifos/objects/user/UserLatLng.java
package com.mifos.objects.user;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Rajan Maurya on 24/01/17.
 */

public class UserLatLng implements Parcelable {

    @SerializedName("lat")
    Double lat;

    @SerializedName("lng")
    Double lng;

    public Double getLat() {
        return lat;
    }

    public void setLat(Double lat) {
        this.lat = lat;
    }

    public Double getLng() {
        return lng;
    }

    public void setLng(Double lng) {
        this.lng = lng;
    }

    @Override
    public String toString() {
        return "{" +
                "lat=" + lat +
                ", lng=" + lng +
                '}';
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(this.lat);
        dest.writeValue(this.lng);
    }

    public UserLatLng(Double latitude, Double longitude) {
        this.lat = latitude;
        this.lng = longitude;
    }

    public UserLatLng() {
    }

    protected UserLatLng(Parcel in) {
        this.lat = (Double) in.readValue(Double.class.getClassLoader());
        this.lng = (Double) in.readValue(Double.class.getClassLoader());
    }

    public static final Parcelable.Creator<UserLatLng> CREATOR =
            new Parcelable.Creator<UserLatLng>() {
        @Override
        public UserLatLng createFromParcel(Parcel source) {
            return new UserLatLng(source);
        }

        @Override
        public UserLatLng[] newArray(int size) {
            return new UserLatLng[size];
        }
    };
}

File: core/datastore/src/main/java/com/mifos/core/objects/user/UserLatLng.kt
package com.mifos.core.objects.user

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

/**
 * Created by Rajan Maurya on 24/01/17.
 */
class UserLatLng : Parcelable {
    @SerializedName("lat")
    var lat: Double = 0.0

    @SerializedName("lng")
    var lng: Double = 0.0
    override fun toString(): String {
        return "{" +
                "lat=" + lat +
                ", lng=" + lng +
                '}'
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(lat)
        dest.writeValue(lng)
    }

    constructor(latitude: Double, longitude: Double) {
        lat = latitude
        lng = longitude
    }

    constructor() {}
    protected constructor(`in`: Parcel) {
        lat = `in`.readValue(Double::class.java.classLoader) as Double
        lng = `in`.readValue(Double::class.java.classLoader) as Double
    }

    companion object {
        @JvmField
        val CREATOR: Creator<UserLatLng> = object : Creator<UserLatLng> {
            override fun createFromParcel(source: Parcel): UserLatLng {
                return UserLatLng(source)
            }

            override fun newArray(size: Int): Array<UserLatLng> {
                return arrayOf()
            }
        }
    }
}